test/quick_test/test_images.py

```diff
@@ -0,0 +1,222 @@
+
+    Returns
+    -------
+    np.ndarray
+        Dice scores for each class.
+    """
+
+    # Classes
+    num_classes = len(labels)
+
+    dscore = np.zeros(shape=num_classes)
+
+    for idx in range(num_classes):
+        current_label = labels[idx]
+
+        pred = (test_data == current_label).astype(int)
+        gt = (reference_data == current_label).astype(int)
+
+        dscore[idx] = dice_score(pred, gt)
+
+    logger.debug("\nDice score: ", dscore)
+
+    return dscore
+
+
+def compute_mean_square_error(test_data, reference_data):
+    """
+    Compute the mean square error between the test and reference data.
+
+    Parameters
+    ----------
+    test_data : np.ndarray
+        Test image data.
+    reference_data : np.ndarray
+        Reference image data.
+
+    Returns
+    -------
+    float
+        Mean square error.
+    """
+
+    mse = ((test_data - reference_data) ** 2).mean()
+    logger.debug("\nMean square error: ", mse)
+
+    return mse
+
+
+@pytest.mark.parametrize("test_subject", load_test_subjects())
+def test_image_headers(subjects_dir: Path, test_dir: Path, reference_dir: Path, test_subject: Path):
+    """
+    Test the image headers by comparing the headers of the test and reference images.
+
+    Parameters
+    ----------
+    subjects_dir : Path
+        Path to the subjects directory.
+    test_dir : Path
+        Name of test directory.
+    reference_dir: Path
+        Name of reference directory.
+    test_subject : Path
+        Name of the test subject.
+
+    Raises
+    ------
+    AssertionError
+        If the image headers do not match
+    """
+
+    # Load images
+    test_subject = subjects_dir / test_dir / test_subject
+    test_image = load_image(test_subject, "brain.mgz")
+
+    reference_subject = subjects_dir / reference_dir / test_subject
+    reference_image = load_image(reference_subject, "brain.mgz")
+
+    # Get the image headers
+    headers = [test_image.header, reference_image.header]
+
+    # Check the image headers
+    header_diff = nibabel.cmdline.diff.get_headers_diff(headers)
+    assert header_diff == OrderedDict(), f"Image headers do not match: {header_diff}"
+    logger.debug("Image headers are correct")
+
+
+@pytest.mark.parametrize("test_subject", load_test_subjects())
+def test_seg_data(subjects_dir: Path, test_dir: Path, reference_dir: Path, test_subject: Path):
+    """
+    Test the segmentation data by calculating and comparing dice scores.
+
+    Parameters
+    ----------
+    subjects_dir : Path
+        Path to the subjects directory.
+    test_dir : Path
+        Name of test directory.
+    reference_dir : Path
+        Name of reference directory.
+    test_subject : Path
+        Name of the test subject.
+
+    Raises
+    ------
+    AssertionError
+        If the dice score is not 0 for all classes
+    """
+
+    test_file = subjects_dir / test_dir / test_subject
+    test_image = load_image(test_file, "aseg.mgz")
```

Comment:
Probably okay for now, but probably we want to test more than just one file, so how do we add more files here?